xarray/tests/test_backends.py

```diff
@@ -1429,6 +1429,172 @@ class TestPyNioAutocloseTrue(TestPyNio):
+                                         })
+            rioda = xr.open_rasterio(tmp_file)
+            assert_allclose(rioda, expected)
+            assert 'crs' in rioda.attrs
+            assert isinstance(rioda.attrs['crs'], basestring)
+
+            # Write it to a netcdf and read again (roundtrip)
+            with create_tmp_file(suffix='.nc') as tmp_nc_file:
+                rioda.to_netcdf(tmp_nc_file)
+                ncds = xr.open_dataarray(tmp_nc_file)
+                assert_identical(rioda, ncds)
+
+    def test_indexing(self):
+
+        import rasterio
+        from rasterio.transform import from_origin
+
+        # Create a geotiff file in latlong proj
+        with create_tmp_file(suffix='.tif') as tmp_file:
+            # data
+            nx, ny, nz = 8, 10, 3
+            data = np.arange(nx*ny*nz,
+                             dtype=rasterio.float32).reshape(nz, ny, nx)
+            transform = from_origin(1, 2, 0.5, 2.)
+            with rasterio.open(
+                    tmp_file, 'w',
+                    driver='GTiff', height=ny, width=nx, count=nz,
+                    crs='+proj=latlong',
+                    transform=transform,
+                    dtype=rasterio.float32) as s:
+                s.write(data)
+            actual = xr.open_rasterio(tmp_file)
+
+            # ref
+            expected = DataArray(data, dims=('band', 'y', 'x'),
+                                 coords={'x': np.arange(nx)*0.5 + 1,
+                                         'y': -np.arange(ny)*2 + 2,
+                                         'band': [1, 2, 3]})
+
+            # tests
+            # assert_allclose checks all data + coordinates
+            assert_allclose(actual, expected)
+
+            # Slicing
+            ex = expected.isel(x=slice(2, 5), y=slice(5, 7))
+            ac = actual.isel(x=slice(2, 5), y=slice(5, 7))
+            assert_allclose(ac, ex)
+
+            ex = expected.isel(band=slice(1, 2), x=slice(2, 5), y=slice(5, 7))
+            ac = actual.isel(band=slice(1, 2), x=slice(2, 5), y=slice(5, 7))
+            assert_allclose(ac, ex)
+
+            # Selecting lists of bands is fine
+            ex = expected.isel(band=[1, 2])
+            ac = actual.isel(band=[1, 2])
+            assert_allclose(ac, ex)
+            ex = expected.isel(band=[0, 2])
+            ac = actual.isel(band=[0, 2])
+            assert_allclose(ac, ex)
+
+            # but on x and y only windowed operations are allowed, more
+            # exotic slicing should raise an error
+            with self.assertRaisesRegexp(IndexError, 'not valid on rasterio'):
+                _ = actual.isel(x=[2, 4], y=[1, 3]).values
+            with self.assertRaisesRegexp(IndexError, 'not valid on rasterio'):
+                _ = actual.isel(x=[4, 2]).values
+            with self.assertRaisesRegexp(IndexError, 'not valid on rasterio'):
+                _ = actual.isel(x=slice(5, 2, -1)).values
+
```

Comment:
Can you add a check that `actual.isel(x=[0, 1], y=[0, 1])` works like `actual.isel(x=slice(2), y=slice(2))`?